test/files/jvm/t2104.scala

```diff
@@ -16,12 +17,13 @@ object Test
   def overwrite(file: JFile, w: FileWriter => Unit): Unit = {
     val fw=new FileWriter(file)
     w(fw)
-    fw.close
-  }
-  def delete_after(f: JFile, g: Source => Unit) = {
-    g(Source.fromFile(f))
-    f.delete
+    fw.close()
   }
+  def delete_after(f: JFile, g: Source => Unit) =
+    Using(Source.fromFile(f)) { src =>
+      g(src)
+      f.delete()
```

Comment:
The delete is supposed to happen after the resource is closed. I wonder if it will fail on windoze...